tests/app/test_jinja_filters.py

```diff
@@ -95,6 +96,28 @@ def test_format_date(self):
 
         self.assertEqual(format_value, '1 January 2017')
 
+    def test_format_conditional_date(self):
+        # Given
+        date = '01/01/2017'
+        date2 = '01/01/2018'
+        dates = ([date, date2])
+
+        # When
+        format_value = format_conditional_date(dates)
+
+        self.assertEqual(format_value, '01 January 2017')
+
+    def test_format_conditional_date2(self):
+        # Given
+        date = None
+        date2 = datetime.strptime('01/01/17', '%d/%m/%y')
+        dates = ([date, date2])
+
+        # When
+        format_value = format_conditional_date(dates)
+
+        self.assertEqual(format_value, '1 January 2017')
+
```

Comment:
If we adapt the method to work with dates in either format for both arguments to `format_conditional_date`, then we'll need additional tests to deal with all of the combinations.